01200_mutations_memory_consumption flakiness

azat opened the pull request:
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

CI:
- https://clickhouse-test-reports.s3.yandex.net/29369/aa8e6014f7a19981c7f7a92eb6d78453d927d461/functional_stateless_tests_flaky_check_(address).html#fail1
- [ClickHouse Functional Stateless Tests Flaky Check (address) for PR #29369](https://clickhouse-test-reports.s3.yandex.net/29369/2320be204397bbb9f9a673cf093b7967fd37a107/functional_stateless_tests_flaky_check_(address).html#fail1)
- https://clickhouse-test-reports.s3.yandex.net/28582/d018cb7adfd516ea290b6b3436b027e5780032de/functional_stateless_tests_(release).html
- https://clickhouse-test-reports.s3.yandex.net/29343/af4e7cd642c0ef7fd1ff93003b223ba49b94a59b/fast_test.html#fail1
- https://clickhouse-test-reports.s3.yandex.net/29369/2320be204397bbb9f9a673cf093b7967fd37a107/functional_stateless_tests_flaky_check_(address)/test_run.txt.out.log

Closes #29415
Introduced-in: #28374
Refs: #18772
Refs: #25165
Refs: #9754 / #9860

azat committed:
Mark 01200_mutations_memory_consumption as long

azat commented:
Can someone add a `force-test` label please (temporary, since I need to get trace-log once the test will fail) ?

azat committed:
Change the build type for flaky stress test to capture trace_log

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=29369) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=29369) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/ClickHouse?pullRequest=29369) it.</sub>

azat commented:
>Functional stateless tests flaky check (address) — Not found changed stateless tests

Incorrectly detects that there are no changes, because this test is new.

azat commented:
>Incorrectly detects that there are no changes, because this test is new (after ec26768)

@alesapin what do you think about adding all new tests for the `Functional stateless tests flaky check (address)`?
It seems that the place where list of such tests is generated is not in repository (so I can't do this by myself)?

azat committed:
Replace 01598_memory_limit_zeros with 01200_mutations_memory_consumption

azat committed:
Add part_log int stateless test artifacts

azat committed:
Move 01200_mutations_memory_consumption out from bugs

azat committed:
Track any allocations

azat committed:
Log peak memory usage after switching

azat committed:
Apply max_untracked_memory/memory_profiler_step during mutate/merge

azat committed:
Set description for memory tracker in MergeList

azat committed:
Reduce number of tests to run on CI

azat committed:
Fix profiler step applying

azat committed:
Fix test name

azat committed:
Fix build for tests

azat committed:
Enable memory sampling (to track free)

azat committed:
Fix memory tracking for mutations/merges

azat committed:
Disable profiler step

azat committed:
Increase timeout for flaky check (to run more iterations)

azat committed:
Removed unused bits from stateless artifacts

It seems that some of steps may take too long and this may fail the
tests with timeout, and I cannot look through artifacts of such tests

azat commented:
Closed in favor of #29691 (plus debugging not via CI)
